### ISSUE TITLE
[BUG] ServiceBusSessionReceiverClient closes only after timeout

### ISSUE THREAD

**Issue by pawel-wilczopolski** [Azure/azure-sdk-for-java#39565]
**Describe the bug**
After invoking ServiceBusSessionReceiverClient `close()` method, the operation lasts for the `tryTimeout` period (1 minute by default) of `AmqpRetryOptions`.
The client should be closed immediately instead of waiting 1 minute.

***Exception or Stack Trace***
The following log is always visible after timeout (default 1 minute) is reached:
`"Timed out waiting for RequestResponseChannel to complete closing. Manually closing."`

**To Reproduce**
1. Accept any session via ServiceBusSessionReceiverClient.
2. Invoke 'close()' method on both ServiceBusSessionReceiverClient and ServiceBusReceiverClient.
3. ServiceBusSessionReceiverClient will be closing 1 minute.

***Code Snippet***

       public Mono<Void> closeAsync(){
            final Mono<Void> closeOperationWithTimeout  = closeMono.asMono().timeout(retryOptions.getTryTimeout()).onErrorResume(TimeoutException.class, error -> {
                return Mono.fromRunnable(() -> {

                    logger.info("Timed out waiting for RequestResponseChannel to complete closing. Manually closing.");

                    onTerminalState("SendLinkHandler");
                    onTerminalState("ReceiveLinkHandler");
                });
            }).subscribeOn(Schedulers.boundedElastic());

        if (isDisposed.getAndSet(true)) {
            logger.verbose("Channel already closed.");
            return closeOperationWithTimeout;
        }

        logger.verbose("Closing request/response channel.");

        return Mono.fromRunnable(() -> {
            try {
                // Schedule API calls on proton-j entities on the ReactorThread associated with the connection.
                provider.getReactorDispatcher().invoke(() -> {
                    logger.verbose("Closing send link and receive link.");

                    sendLink.close();
                    receiveLink.close();
                });
            } catch (IOException | RejectedExecutionException e) {
                logger.info("Unable to schedule close work. Closing manually.");

                sendLink.close();
                receiveLink.close();
            }
        }).subscribeOn(Schedulers.boundedElastic()).then(closeOperationWithTimeout);
    

**Expected behavior**
The ServiceBusSessionReceiverClient should be closed immediately.

**Setup (please complete the following information):**
 - OS: Windows 11
 - IDE: IntelliJ IDEA
 - Library/Libraries: azure-messaging-servicebus:7.15.2
 - Java version: 11

**Comment by anuchandy** [comment]
Hello @pawel-wilczopolski hope you're doing good. @conniey and I discussed this today. So, the internal state machine of AMQP library is asynchronous and the reason for it wait on force close is to ensure any lingering messages, dispositions (complete|abandon) enqueued and other state in transit is processed, so things are cleaned up gracefully.
 
If a continuous receive and high throughput is desired, it is recommended to use the ServiceBusProcessorClient. Please see [here](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/servicebus/azure-messaging-servicebus#when-to-use-servicebusprocessorclient). As it says there, the use of Synchronous Client (ServiceBus[Session|Receiver]Client) is typically discouraged, because the user has to deal with more errors and possible message expirations. On the other hand, the ServiceBusProcessorClient handles errors automatically and takes care of multi-threading, so receive from multiple sessions can happen in parallel. For the long term, ServiceBusProcessorClient will help the applications be more reliable.
 
If Synchronous client is still needed, and if you’re going to build a new ServiceBusSessionClient instance or restart the service after the close, you may do a fire-and-forget close, i.e, invoke close on a separate thread and giving it a few seconds before the restart/building new client using a new builder. The code may looks like-
```java
// The ServiceBusSessionReceiver is no longer needed.
// Fire and forget the client close
try {
CompletableFuture.supplyAsync(() -> {
    // call ServiceBusSessionReceiver::close.
return null;
})
.toCompletableFuture().get(5, TimeUnit.SECONDS);
} catch (Exception e) {
    // log(e);
}
// Service restarts or application creates new ServiceBusSessionReceiver
```

**Comment by pawel-wilczopolski** [comment]
Hi all,

I cannot use ServiceBusProcessorClient because it hides a lot of errors that I need to handle on my side. Your solution is not resolving the real issue. I found out that you have a bug because the waiting logic begins too fast.
In other words – we start waiting 1 minute BEFORE we get the ACK from the server that would cause the graceful shutdown immediately. Please refer to this.

Regards,
Paweł Wilczopolski

**Comment by anuchandy** [comment]
Adding additional notes on the safety of the [above workaround](https://github.com/Azure/azure-sdk-for-java/issues/39565#issuecomment-2048553634) provided - when offloading the closure with a client-side timeout, there are two execution paths, both leading to eventual shutdown and garbage collection of reactor-executor thread, i.e., the reactor-executor thread won’t stay around and will not impact the production run.
•	[Execution-Path-1] Either application or host (VM, container) exit after calling close().
•	[Execution-Path-2] Or, the application calls close() and continues with creating a new Client.
In the first case, since the JVM exits, all threads get cleaned up as part of the exit. In the second case, the old reactor-executor thread will continue its final cleanup activities on the background (without blocking the main thread) and exit.

**We’ve created a work item to improve this experience. Linking it [here](https://github.com/Azure/azure-sdk-for-java/issues/39876), please track it there. Closing this.**
